configs/templates/gsi-addon-dev/spack.yaml

```diff
@@ -0,0 +1,25 @@
+spack:
+  concretizer:
+    unify: when_possible
+
+  view: false
+  include:
+  - site
+  - common
+  definitions:
+  # Note: Set 'compilers' manually; must match upstream list
+  - compilers: []
```

Comment:
Maybe in a future release of spack-stack, we can automate this with the `spack-stack create env --upstream` command? Infer the compiler list from the first upstream repo provided and set it here.